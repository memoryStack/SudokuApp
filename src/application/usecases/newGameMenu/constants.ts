// TODO: how to compose enums so that's it's easier to edit or re-organize them
// TODO: these names are confusing, fix these
export enum AUTO_GENERATED_NEW_GAME_IDS {
    EASY = 'EASY',
    MEDIUM = 'MEDIUM',
    HARD = 'HARD',
    EXPERT = 'EXPERT',
}

export enum NEW_GAME_IDS {
    EASY = 'EASY',
    MEDIUM = 'MEDIUM',
    HARD = 'HARD',
    EXPERT = 'EXPERT',
    CUSTOMIZE_PUZZLE = 'CUSTOMIZE_PUZZLE',
}

export enum START_GAME_MENU_ITEMS_IDS {
    EASY = 'EASY',
    MEDIUM = 'MEDIUM',
    HARD = 'HARD',
    EXPERT = 'EXPERT',
    CUSTOMIZE_PUZZLE = 'CUSTOMIZE_PUZZLE',
    RESUME = 'RESUME'
}
