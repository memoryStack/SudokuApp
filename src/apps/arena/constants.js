import { roundToNearestPixel } from '../../utils/util'

export const BOARD_AXES_VALUES = {
    Y_AXIS: ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I'],
    X_AXIS: ['1', '2', '3', '4', '5', '6', '7', '8', '9'],
}

export const BOARD_CELLS_COUNT = 81
export const MAX_INSTANCES_OF_NUMBER = 9
export const CELLS_IN_HOUSE = 9
export const NUMBERS_IN_HOUSE = 9
export const HOUSES_COUNT = 9
export const BOARD_TO_WINDOW_WIDTH_RATIO = 0.94
export const THICK_BORDER_COUNT = 2
export const THIN_BORDER_COUNT = 8
export const BLOCKS_COUNT_IN_ROW = 3

export const GRID_TRAVERSALS = {
    ROW: 'ROW',
    COL: 'COL',
    BLOCK: 'BLOCK',
}

export const STATIC_BOARD_ELEMENTS_DIMENSIONS = {
    THICK_BORDER_WIDTH: roundToNearestPixel(3),
    THIN_BORDER_WIDTH: roundToNearestPixel(1),
    AXIS_WIDTH: roundToNearestPixel(20),
}

export const DEEPLINK_PUZZLE_URL_ERRORS = {
    EMPTY_URL: 'Invaild Url',
    INCOMPLETE_NUMBERS: `Puzzle doesn't have ${BOARD_CELLS_COUNT} numbers`,
    INVALID_CHARACTERS: 'Puzzle has invalid characters other than 0-9',
}

export const PUZZLE_SOLUTION_TYPES = {
    NO_SOLUTION: 'NO_SOLUTION',
    UNIQUE_SOLUTION: 'UNIQUE_SOLUTION',
    MULTIPLE_SOLUTIONS: 'MULTIPLE_SOLUTIONS',
}

export const BOARD_MOVES_TYPES = {
    ADD: 'ADD',
    REMOVE: 'REMOVE',
}

export const BOARD_GRID_BORDERS_DIRECTION = {
    VERTICAL: 'vertical',
    HORIZONTAL: 'horizontal',
}

export const ARENA_PAGE_TEST_ID = 'arena_test_id'

export const GAME_OVER_CARD_OVERLAY_TEST_ID = 'GAME_OVER_CARD_OVERLAY_TEST_ID'
