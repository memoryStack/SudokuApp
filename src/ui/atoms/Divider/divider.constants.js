export const DIVIDER_TYPES = {
    HORIZONTAL: 'HORIZONTAL',
    VERTICAL: 'VERTICAL',
}

export const DIVIDER_TEST_ID = 'divider_test_id'
