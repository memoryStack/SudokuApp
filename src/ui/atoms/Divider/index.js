export { default } from './Divider'
export * from './divider.constants'
