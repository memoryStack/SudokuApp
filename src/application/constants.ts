import { AUTO_GENERATED_NEW_GAME_IDS } from "./usecases/newGameMenu/constants"

export enum BOARD_MOVES_TYPES {
    ADD = 'ADD',
    REMOVE = 'REMOVE',
}

export enum PENCIL_STATE {
    ACTIVE = 'ACTIVE',
    INACTIVE = 'INACTIVE',
}

export enum GAME_STATE {
    ACTIVE = 'ACTIVE',
    INACTIVE = 'INACTIVE',
    DISPLAY_HINT = 'DISPLAY_HINT',
    GAME_SELECT = 'GAME_SELECT',
    OVER_SOLVED = 'SOLVED',
    OVER_UNSOLVED = 'UNSOLVED',
}

export enum LEVEL_DIFFICULTIES {
    EASY = 'EASY',
    MEDIUM = 'MEDIUM',
    HARD = 'HARD',
    EXPERT = 'EXPERT',
}

export const GAME_LEVEL_VS_CLUES = {
    [AUTO_GENERATED_NEW_GAME_IDS.EASY]: 36,
    [AUTO_GENERATED_NEW_GAME_IDS.MEDIUM]: 32,
    [AUTO_GENERATED_NEW_GAME_IDS.HARD]: 28,
    [AUTO_GENERATED_NEW_GAME_IDS.EXPERT]: 24,
}
