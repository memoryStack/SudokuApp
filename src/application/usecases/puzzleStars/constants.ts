import { NEW_GAME_IDS } from "../newGameMenu/constants"

const TIME_LIMIT_FOR_ONE_STAR = 100000000

// this timeLimit is in minutes
export const PUZZLE_STARS_DISTRIBUTION = {
    [NEW_GAME_IDS.EASY]: [{ stars: 3, timeLimit: 20 }, { stars: 2, timeLimit: 30 }, { stars: 1, timeLimit: TIME_LIMIT_FOR_ONE_STAR }],
    [NEW_GAME_IDS.MEDIUM]: [{ stars: 3, timeLimit: 30 }, { stars: 2, timeLimit: 40 }, { stars: 1, timeLimit: TIME_LIMIT_FOR_ONE_STAR }],
    [NEW_GAME_IDS.HARD]: [{ stars: 3, timeLimit: 40 }, { stars: 2, timeLimit: 50 }, { stars: 1, timeLimit: TIME_LIMIT_FOR_ONE_STAR }],
    [NEW_GAME_IDS.EXPERT]: [{ stars: 3, timeLimit: 50 }, { stars: 2, timeLimit: 60 }, { stars: 1, timeLimit: TIME_LIMIT_FOR_ONE_STAR }]
}
