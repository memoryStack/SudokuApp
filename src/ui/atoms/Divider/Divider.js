import React from 'react'

import { View } from 'react-native'

import PropTypes from 'prop-types'

import { useStyles } from '@utils/customHooks/useStyles'

import { getStyles } from './divider.styles'
import { DIVIDER_TEST_ID, DIVIDER_TYPES } from './divider.constants'

const Divider = ({
    style,
    type,
    ...rest
}) => {
    const styles = useStyles(getStyles, { type })

    return (
        <View
            testID={DIVIDER_TEST_ID}
            style={[styles.container, style]}
            {...rest}
        />
    )
}

export default React.memo(Divider)

Divider.propTypes = {
    style: PropTypes.object,
    type: PropTypes.oneOf(Object.values(DIVIDER_TYPES)),
}

Divider.defaultProps = {
    style: null,
    type: DIVIDER_TYPES.HORIZONTAL,
}
