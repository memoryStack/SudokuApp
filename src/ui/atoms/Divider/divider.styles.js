import { StyleSheet } from 'react-native'

import _get from '@lodash/get'

import { roundToNearestPixel } from '@utils/util'

import { DIVIDER_TYPES } from './divider.constants'

export const getStyles = ({ type }, theme) => {
    const isHorizontal = type === DIVIDER_TYPES.HORIZONTAL
    const crossDirectionDimensionProperty = isHorizontal ? 'height' : 'width'
    const inDirectionDimensionProperty = isHorizontal ? 'width' : 'height'

    return StyleSheet.create({
        container: {
            [inDirectionDimensionProperty]: '100%',
            [crossDirectionDimensionProperty]: roundToNearestPixel(_get(theme, ['divider', 'thickness'])),
            backgroundColor: _get(theme, ['divider', 'color']),
        },
    })
}
