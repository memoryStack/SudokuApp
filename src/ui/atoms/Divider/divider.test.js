import * as React from 'react'

import { render, screen } from '@utils/testing/testingLibrary'

import Divider from './Divider'
import { DIVIDER_TEST_ID, DIVIDER_TYPES } from './divider.constants'

test('snapshot for vertical divider', () => {
    const tree = render(
        <Divider type={DIVIDER_TYPES.VERTICAL} />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
})

test('snapshot for horizontal divider', () => {
    const tree = render(
        <Divider type={DIVIDER_TYPES.HORIZONTAL} />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
})

test('Styles Overrides', () => {
    const customStyles = { backgroundColor: 'some random color' }
    render(
        <Divider
            type={DIVIDER_TYPES.HORIZONTAL}
            style={customStyles}
        />,
    )

    expect(screen.getByTestId(DIVIDER_TEST_ID)).toHaveStyle(customStyles)
})
