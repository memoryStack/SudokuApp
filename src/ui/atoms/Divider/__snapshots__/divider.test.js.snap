// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot for horizontal divider 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#CAC4D0",
        "height": 1,
        "width": "100%",
      },
      null,
    ]
  }
  testID="divider_test_id"
/>
`;

exports[`snapshot for vertical divider 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#CAC4D0",
        "height": "100%",
        "width": 1,
      },
      null,
    ]
  }
  testID="divider_test_id"
/>
`;
